File: mario.c
#include <cs50.h>
#include <stdio.h>

void print_offset(int space);
void print_row(int height);

const int n;

int main(void)
{
    int n = get_int("Height? ");
    // for (int i = 0; i < n; i++)
    while ( n > 0)
    {
        for (int j = 0; j < n ; n--)
        {
            print_offset(n - 1);
        }
        
    }
}

void print_row(int length)
{
    for (int i = 0; i < length; i++)
    {
        printf("#");
    }
    printf("\n");
}

void print_offset(int space)
{
    for (int j = 0; j < space; j++)
    {
        printf(" ");
    }
    print_row(space);
}

File: mario/mario.py
from cs50 import get_int


def get_height(prompt):
    while True:
        try:
            height = int(input(prompt))
            if 2 <= height <= 8:
                return height
            else:
                print("Please select a number between 2 and 8 inclusive.")
        except ValueError:
            print("Please select a number between 2 and 8 inclusive.")


def main():
    height = get_height("Height? ")
    for x in range(height, 0, -1):
        print(" " * x, end="")
        print("#" * (height - x + 1))


main()
